### ISSUE TITLE
TLS 1.3

### ISSUE THREAD

**Issue by jibanes** [Lochnair/xt_tls#56]
I recall that TLS 1.3 has a (non-mandatory) option to encrypt the SNI field, is it reasonable to assume that this isn't something that this module would be able to handle?

**Comment by Lochnair** [comment]
Yeah, connections with encrypted SNI is not something this module can handle, now or ever.

It's been a while since I've read about this, but given ECH (encrypted client header) is a thing, it's likely that in some undetermined number of years in the future, this module will be of no use anymore.

**Comment by jibanes** [comment]
Or let's just say that when this becomes a reality, if it ever does, it
will require a bit more magic, meanwhile this is extremely valuable.

**Comment by Lochnair** [comment]
I'm assuming it will be in the somewhat near future, but for now this will keep functioning.

**Comment by jibanes** [comment]
So essentially it will be mostly for browsers as I understand, for privacy
reasons (possibly to escape govt surveillance), but it's probably somewhat
unlikely that applications relying exclusively on openssl libraries will
leverage that, at least for the next few years?
